doc/_scripts/merge_search_indexes.py

```diff
@@ -0,0 +1,214 @@
+"""
+Sphinx search index merge utility
+=================================
+
+This script can be used to merge search indexes of multiple docsets, thus
+allowing to obtain a cross-search experience.
+
+Credits go to Dominik Kilian for the original idea and part of this code.
```

Comment:
I don't think that this line is required. You mentioned this in the PR. Optionally, you can add `Co-authored-by:` in the commit message.